### DISCUSSION TITLE
how to create multiple forms based on an array on 1 page

### DISCUSSION THREAD

**Discussion by Felixaverlant** [react-hook-form/react-hook-form#8739] [Q&A]
Hi there,

Thanks for the amazing work.

I'm currently working on a project where I need to generate forms based on an array.

As of today, if I submit form 1, form 2 gets autocomplete with value of form 1.
Ex:
```js
const arrayForm = [
 {name: 'form 1', id: '1'},
 {name: 'form 2', id: '2'},
] 

const FormComp = ({id}) => {
  const {register} = useForm()
  return (
     <input /> 
     <submit />
  )
}

arrayForm.map(form => {
  <FormComp id={id} />
}
```

## Idea 1: rename register
I thought I could rename the register dynamically but that's impossible ( https://stackoverflow.com/questions/60276510/how-to-make-react-hook-form-work-with-multiple-forms-in-one-page)

## Idea 2 : unique register value
Doing `register('name1-input1')` doesn't work either. Same result.

## Idea 3 : using a "fake" multistep form
Not tested so feedbacks welcome.

## Idea 4 : key on component
I tried implementing this : https://stackoverflow.com/questions/66831001/react-hook-form-register-different-forms-with-same-field-names
But it doesn't seems to work and is for multistep form anyway

Thank for all your help!

**Comment by sp88011** [comment]
I'm struggling with the same problem. Would appreciate any pointers. Idea 1 would be the cleanest...if only it worked.

**Comment by Felixaverlant** [comment]
I wrapped the library and used an id for each form and fields. Hope it helped!
